### DISCUSSION TITLE
Should we use LIME for text function and modify it for SMILE?

### DISCUSSION THREAD

**Discussion by mojgan1987** [Dependable-Intelligent-Systems-Lab/xwhy#2] [Q&A] [answered]
The use of LIME for text explainability has shown promise in providing insightful interpretations of complex text models. However, the standard LIME method relies on distances that may not fully capture the structure of text data. Modifying the distances function to SMILE-based distances, such as Wasserstein, can result in an improvement in the quality of the text explainability.

This raises the question of whether it is more efficient to modify the existing LIME function or to create a new function specifically designed for SMILE-based text explainability. While modifying the LIME function may be faster in the short term, it may not fully leverage the strengths of SMILE-based distances. On the other hand, writing a new function from scratch can result in a more robust and flexible solution, tailored specifically to the unique challenges of text explainability.

Ultimately, the decision between modifying the existing LIME function or writing a new function will depend on the specific goals and constraints of the project. In any case, a thorough evaluation of the resulting explanations and an assessment of their effectiveness in meeting the project requirements is recommended.

**Comment by koo-ec** [comment, marked as answer]
I completely agree with the idea of using existing LIME functions and modifying them for SMILE. Modifying the existing LIME functions can save a lot of time and effort in the development process, while still allowing us to take advantage of the strengths of SMILE-based distances. This approach can be a good starting point and can provide useful results, especially if the goal is to quickly obtain interpretable explanations.

However, it's important to remember that the modified LIME function may not fully capture the unique characteristics of the SMILE method, and it may need to be further customized or optimized based on the specific requirements of the project. Additionally, a thorough evaluation of the modified LIME function and its results is crucial to ensure that it meets the desired goals and objectives.
